bot/cogs/help.py

```diff
@@ -70,499 +68,307 @@ class HelpSession:
-        await self.message.delete()
-
-
-class Help(DiscordCog):
-    """Custom Embed Pagination Help feature."""
+        lines = []
 
-    @commands.command('help')
-    @redirect_output(destination_channel=Channels.bot, bypass_roles=STAFF_ROLES)
-    async def new_help(self, ctx: Context, *commands) -> None:
-        """Shows Command Help."""
-        try:
-            await HelpSession.start(ctx, *commands)
-        except HelpQueryNotFound as error:
-            embed = Embed()
-            embed.colour = Colour.red()
-            embed.title = str(error)
+        for cog_or_category, _commands in itertools.groupby(filter_commands, key=self._category_key):
+            sorted_commands = sorted(_commands, key=lambda c: c.name)
 
-            if error.possible_matches:
-                matches = '\n'.join(error.possible_matches.keys())
-                embed.description = f'**Did you mean:**\n`{matches}`'
+            if len(sorted_commands) == 0:
+                continue
 
-            await ctx.send(embed=embed)
+            fmt = [
+                f"`{PREFIX}{c.qualified_name}{f' {c.signature}' if c.signature else ''}`"
+                f"\n*{c.short_doc or 'No details provided.'}*" for c in sorted_commands
+            ]
 
+            # we can't embed a '\n'.join() inside an f-string so this is a bit of a compromise
+            def get_fmt(i: int) -> str:
+                """Get a formatted version of commands for an index."""
+                return "\n".join(fmt[i:i+COMMANDS_PER_PAGE])
 
-def unload(bot: Bot) -> None:
-    """
-    Reinstates the original help command.
-
-    This is run if the cog raises an exception on load, or if the extension is unloaded.
-    """
-    bot.remove_command('help')
-    bot.add_command(bot._old_help)
-
+            # this is a bit yuck because moderation category has 8 commands which needs to be split over 2 pages.
+            # pretty much it only splits that category, but also gives the number of commands it's adding to
+            # the pages every iteration so we can easily use this below rather than trying to split the string.
+            lines.extend(
+                (
+                    (f"**{cog_or_category}**\n{get_fmt(i)}", len(fmt[i:i+COMMANDS_PER_PAGE]))
+                    for i in range(0, len(sorted_commands), COMMANDS_PER_PAGE)
+                )
+            )
 
-def setup(bot: Bot) -> None:
-    """
-    The setup for the help extension.
+        pages = []
+        counter = 0
+        formatted = ""
+        for (fmt, length) in lines:
+            counter += length
+            if counter > COMMANDS_PER_PAGE:
+                # force a new page on paginator even if it falls short of the max pages
+                # since we still want to group categories/cogs.
+                counter = length
+                pages.append(formatted)
+                formatted = f"{fmt}\n\n"
+                continue
+            formatted += f"{fmt}\n\n"
```

Comment:
I prefer `else` over `continue` in most cases as it's easier to understand. In this case, only one line would go under `else`. I use `continue` when there's a lot of code still below.